xray/plot/plot.py

```diff
@@ -424,27 +438,63 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
         if ax is None:
             ax = plt.gca()
 
-        # Handle the dimensions
-        try:
-            ylab, xlab = darray.dims
-        except ValueError:
-            raise ValueError('{name} plots are for 2 dimensional DataArrays. '
+        dims = list(darray.dims)
+
+        if len(dims) != 2:
+            raise ValueError('{type} plots are for 2 dimensional DataArrays. '
                              'Passed DataArray has {ndim} dimensions'
-                             .format(name=plotfunc.__name__, ndim=len(darray.dims)))
+                             .format(type=plotfunc.__name__, ndim=len(dims)))
+
+        if x and x not in darray:
+            raise KeyError('{0} is not a dimension of this DataArray. Use '
```

Comment:
I'm a bit confused here, so correct me if I'm mistaken. Dimensions are the labels corresponding to the axes. I do actually want to test on this. Which implies I should instead do `if x and x not in dims`.

You're saying that `x not in darray` is equivalent to `x not in darray.coords`, right?